float32 X
float32 Y
float32 Th
float32 pathDistance